src/unix/tty.c

```diff
@@ -113,6 +114,17 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
     fd = newfd;
   }
 
+  /* save the fd flags in case we need to restore them due to an error */
+  saved_flags = fcntl(fd, F_GETFL);
```

Comment:
You should put that in 'while EINTR' loop for robustness.  `fcntl(F_GETFL)` isn't really interruptable but I'd err on the side of caution.

Also, capitalized and punctuated comments, please! =)